simple/msg_pingpong.c

```diff
@@ -621,14 +567,23 @@ int main(int argc, char **argv)
 	if (optind < argc)
 		dst_addr = argv[optind];
 
-	hints.domain_attr = &domain_hints;
-	hints.ep_attr = &ep_hints;
+	if (opts.src_addr) {
+		ret = getaddr(opts.src_addr, opts.port,
+				(struct sockaddr **) &hints.src_addr,
+				(socklen_t *) &hints.src_addrlen);
+
+		if (ret) {
+			FI_DEBUG("source address error %s\n", gai_strerror(ret));
+			return EXIT_FAILURE;
+		}
+	}
+
```

Comment:
I had it like that but it requires that the -p is specified on the command line before -s , which is not OK. What about moving that snippet into getaddr so it becomes:
    ft_getaddr(char *addr, char *port, struct fi_info *hints)

?